[CI] MixedClusterClientYamlTestSuiteIT test {p0=synonyms/90_synonyms_reloading_for_synset/Reload analyzers for specific synonym set} failing

**Build Scans:**
- [elasticsearch-intake #12497 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/j77ul3nsmqnag)
- [elasticsearch-pull-request #40100 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/i3fp7evbfmiig)
- [elasticsearch-pull-request #40090 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/nxfq2bapv3kem)
- [elasticsearch-pull-request #40083 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/m7bmp62b3tpd4)
- [elasticsearch-pull-request #40065 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/aifr5fyrbr4zm)
- [elasticsearch-pull-request #39825 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/cqtuygbj45wfy)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.17.0#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=synonyms/90_synonyms_reloading_for_synset/Reload analyzers for specific synonym set}" -Dtests.seed=F4CD5A9863D960CB -Dtests.bwc=true -Dtests.locale=zh-SG -Dtests.timezone=America/Metlakatla -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dsynonyms%2F90_synonyms_reloading_for_synset%2FReload%20analyzers%20for%20specific%20synonym%20set%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [synonyms/90_synonyms_reloading_for_synset:125]: 
Expected: <1>
     but: was <0>
```

**Issue Reasons:**
- [main] 6 failures in test test {p0=synonyms/90_synonyms_reloading_for_synset/Reload analyzers for specific synonym set} (3.0% fail rate in 199 executions)
- [main] 6 failures in step 8.17.0_bwc-snapshots (3.3% fail rate in 183 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (3.8% fail rate in 132 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -192,9 +192,6 @@ tests:
 - class: org.elasticsearch.action.search.SearchQueryThenFetchAsyncActionTests
   method: testBottomFieldSort
   issue: https://github.com/elastic/elasticsearch/issues/116249
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=synonyms/90_synonyms_reloading_for_synset/Reload analyzers for specific synonym set}
-  issue: https://github.com/elastic/elasticsearch/issues/116332
 - class: org.elasticsearch.xpack.shutdown.NodeShutdownIT
   method: testAllocationPreventedForRemoval
   issue: https://github.com/elastic/elasticsearch/issues/116363

